controllers/datadogagent/feature/enabledefault/feature.go

```diff
@@ -233,18 +252,19 @@ func (f *defaultFeature) ManageDependencies(managers feature.ResourceManagers, c
 	// manage credential secret
 	if f.credentialsInfo.secretCreation.createSecret {
 		for key, value := range f.credentialsInfo.secretCreation.data {
-			if err := managers.SecretManager().AddSecret(f.owner.GetNamespace(), f.credentialsInfo.secretCreation.name, key, value); err != nil {
+			if err := managers.SecretManager().AddSecret(f.logger, f.owner.GetNamespace(), f.credentialsInfo.secretCreation.name, key, value, nil); err != nil {
 				errs = append(errs, err)
 			}
 		}
 	}
 
 	if components.ClusterAgent.IsEnabled() && f.dcaTokenInfo.secretCreation.createSecret {
 		for key, value := range f.dcaTokenInfo.secretCreation.data {
-			if err := managers.SecretManager().AddSecret(f.owner.GetNamespace(), f.dcaTokenInfo.secretCreation.name, key, value); err != nil {
+			if err := managers.SecretManager().AddSecret(f.logger, f.owner.GetNamespace(), f.dcaTokenInfo.secretCreation.name, key, value, map[string]string{f.customConfigAnnotationKey: f.customConfigAnnotationValue}); err != nil {
```

Comment:
I'm wondering if it should be more clear to add a dedicated function  `managers.SecretManager().AddAnnotation()` to keep simple the `AddSecret()` function